bot/exts/moderation/voice_gate.py

```diff
@@ -0,0 +1,150 @@
+
+    @command(aliases=('voiceverify', 'vverify', 'voicev', 'vv'))
+    @has_no_roles(Roles.voice_verified)
+    @in_whitelist(channels=(Channels.voice_gate,), redirect=None)
+    async def voice_verify(self, ctx: Context, *_) -> None:
+        """
+        Apply to be able to use voice within the Discord server.
+
+        In order to use voice you must meet all three of the following criteria:
+        - You must have over a certain number of messages within the Discord server
+        - You must have accepted our rules over a certain number of days ago
+        - You must not be actively banned from using our voice channels
+        """
+        # Send this as first thing in order to return after sending DM
+        await ctx.send("You will get response to DM.")
+
+        try:
+            data = await self.bot.api_client.get(f"bot/users/{ctx.author.id}/metricity_data")
+        except ResponseCodeError as e:
+            if e.status == 404:
+                embed = discord.Embed(
+                    title="Not found",
+                    description=f"{ctx.author.mention} Unable to find Metricity data about you.",
+                    color=Colour.red()
+                )
+                log.info(f"Unable to find Metricity data about {ctx.author} ({ctx.author.id})")
+            else:
+                embed = discord.Embed(
+                    title="Unexpected response",
+                    description="Got unexpected response from site. Please let us know about this.",
+                    color=Colour.red()
+                )
+                log.warning(f"Got response code {e.status} while trying to get {ctx.author.id} Metricity data.")
+
+            await ctx.author.send(embed=embed)
+            return
+
+        # Pre-parse this for better code style
+        if data["verified_at"] is not None:
+            data["verified_at"] = parser.isoparse(data["verified_at"])
+        else:
+            data["verified_at"] = datetime.utcnow() - timedelta(days=3)
+
+        checks = {
+            "verified_at": data["verified_at"] > datetime.utcnow() - timedelta(days=GateConf.minimum_days_verified),
+            "total_messages": data["total_messages"] < GateConf.minimum_messages,
+            "voice_banned": data["voice_banned"]
+        }
+        failed = any(checks.values())
+        failed_reasons = [MESSAGE_FIELD_MAP[key] for key, value in checks.items() if value is True]
+        [self.bot.stats.incr(f"voice_gate.failed.{key}") for key, value in checks.items() if value is True]
+
+        if failed:
+            if len(failed_reasons) > 1:
+                reasons = f"{', '.join(failed_reasons[:-1])} and {failed_reasons[-1]}"
+            else:
+                reasons = failed_reasons[0]
+
+            embed = discord.Embed(
+                title="Voice Gate not passed",
+                description=FAILED_MESSAGE.format(user=ctx.author.mention, reasons=reasons),
+                color=Colour.red()
+            )
```

Comment:
Let's update this so it is prettier in embeds. How about this:
<img width="599" alt="Screenshot 2020-10-18 at 12 32 21" src="https://user-images.githubusercontent.com/20439493/96366209-fb327f80-113d-11eb-9e38-999ff91ab428.png">